### PR TITLE
Pimpl'ize ogre2 classes

### PR DESCRIPTION
**Pull request by iche033**
Updated several ogre2 classes to use pimpl. Most of the private classes are currently empty.

Signed-off-by: Ian Chen <ichen@osrfoundation.org>

### CODE REVIEW

File: ogre2/include/ignition/rendering/ogre2/Ogre2GaussianNoisePass.hh [hunk 1]
```diff
@@ -47,8 +50,8 @@ namespace ignition
       // Documentation inherited
       public: void CreateRenderPass() override;
 
-      /// brief Pointer to the Gaussian noise ogre material
-      private: Ogre::Material *gaussianNoiseMat = nullptr;
+      /// \brief Pointer to private data class
+      private: std::unique_ptr<Ogre2GaussianNoisePassPrivate> dataPtr;
```
Comment (chapulina) [inline ogre2/include/ignition/rendering/ogre2/Ogre2GaussianNoisePass.hh:54]: I didn't see a warning, but there's no include memory.

File: ogre2/src/Ogre2Camera.cc [hunk 2]
```diff
@@ -24,11 +24,17 @@
...
+class ignition::rendering::Ogre2CameraPrivate
+{
+};
+
 using namespace ignition;
 using namespace rendering;
 
 //////////////////////////////////////////////////
 Ogre2Camera::Ogre2Camera()
+  : dataPtr(new Ogre2CameraPrivate)
```
Comment (chapulina) [inline ogre2/src/Ogre2Camera.cc:37]: Consider using `make_unique` for all `dataPtr`s

**Review by chapulina — APPROVED** [review]
LGTM, I just have some smaller comments that I'll leave up to you.

File: ogre2/include/ignition/rendering/ogre2/Ogre2GaussianNoisePass.hh (hunk 1 above)
Comment (iche033) [reply ogre2/include/ignition/rendering/ogre2/Ogre2GaussianNoisePass.hh:56]: added. d2b2979df0ced5ac848da30d8ecb021084c7ea53

File: ogre2/src/Ogre2Camera.cc (hunk 2 above)
Comment (iche033) [reply ogre2/src/Ogre2Camera.cc]: changed to `make_unique` d2b2979df0ced5ac848da30d8ecb021084c7ea53

**Pull request merged**